Include Share in iOS SystemContextMenu defaults

muradhossin opened the pull request:
This PR fixes #173491 by adding the missing 'Share' option to the default iOS SystemContextMenu when `shareEnabled` is true.

**Changes:**
- Added `IOSSystemContextMenuItemShare` to `getDefaultItems` in `system_context_menu.dart`.
- Added a widget test to ensure Share is present in the default items for non-empty selections on iOS.

**Rationale:**
This aligns Flutter's default iOS text selection context menu with native iOS behavior, ensuring users see the expected 'Share' option when selecting text.

**Demo:**
Video showing Share option in iOS context menu: https://github.com/user-attachments/assets/e04cd1f9-7d92-4147-a09b-719f03d9c625

muradhossin committed:
feat(cupertino): include Share in iOS SystemContextMenu defaults

Fixes #173491.

This adds IOSSystemContextMenuItemShare to the default iOS system
context menu items when shareEnabled is true, ensuring native 'Share'
option appears for text selections.

justinmc commented:
@hellohuanlin Did we purposely not include Share, or is this ok to add?

justinmc commented:
Thanks for the quick response @hellohuanlin!

justinmc commented on packages/flutter/test/widgets/system_context_menu_test.dart:
```diff
@@ -895,4 +895,57 @@ void main() {
+      cursorColor: Colors.blue,
+      backgroundCursorColor: Colors.grey,
+      // Use the system context menu pipeline on iOS.
+      contextMenuBuilder: (BuildContext _, EditableTextState s) {
```
Don't use underscore for parameter names and don't abbreviate (`s`) per the [styleguide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md).

justinmc commented on packages/flutter/test/widgets/system_context_menu_test.dart:
```diff
@@ -895,4 +895,57 @@ void main() {
+    expect(items.any((e) => e is IOSSystemContextMenuItemShare), isTrue);
+
+    debugDefaultTargetPlatformOverride = null;
+  });
```
How I would test this: Listen to 'ContextMenu.showSystemContextMenu' (see elsewhere in the file for how to do that), then show the context menu and expect that it was called with a "share" item.

justinmc reviewed (commented):
Some suggestions for the test, but otherwise thank you for noticing this and opening up a PR!

muradhossin committed:
test: iOS system context menu test follows framework style (inline widget, iOS variant, channel assert)

muradhossin commented:
Demo video has been added.

- Removed debugDefaultTargetPlatformOverride (confirmed not present in the file).
- Removed _HostEditable; inlined the widget tree directly in the test.
- No extraneous controller/focusNode usage — only a controller is used, and it’s disposed via addTearDown.
- Dropped unused cursorColor / backgroundCursorColor parameters.
- Used descriptive parameter names in contextMenuBuilder.
- Test now listens to ContextMenu.showSystemContextMenu and asserts that a Share item is present.
- No platform override resets are required.

justinmc commented on packages/flutter/test/widgets/system_context_menu_test.dart:
```diff
@@ -895,4 +895,71 @@ void main() {
+      await tester.pump();
+
+      // Assert that the platform message included a Share item.
+      expect(itemsReceived, isNotEmpty);
```
Nit: Can you add an expectation that it *is* empty before the toolbar is shown?

justinmc reviewed (approved):
LGTM with a nit 👍 Thanks for fixing my comments and for adding the demo video.

ValentinVignal reviewed (approved):
LGTM

muradhossin committed:
test: verify no platform message is sent before showing the iOS context menu toolbar

muradhossin commented:
@justinmc I’ve rebased fix/ios-share-context-menu onto the latest master, resolved conflicts, and pushed (analyze + test pass locally). Could you please reopen #173626 and merge? Thank you! 🙏

muradhossin commented:
Hi @justinmc  

Thanks again for the earlier review and approval. I’ve rebased this branch and confirmed that all checks are passing. Since the PR was closed, could you please re-open it for merge?

If re-opening isn’t possible, I’m happy to create a fresh PR and reference this one (#173626) along with the linked issue (#173491). Just let me know what works best. 🙏

Thanks for your guidance and support!
